core/src/element.rs

```diff
@@ -77,6 +77,15 @@ impl Element {
             _ => None,
         }
     }
+
+    /// Get the name of a element (if it has one).
+    pub fn name(&self) -> Option<&String> {
```

Comment:
Changing it to `&str`. And yes it's a good point that it might be a bit unnecessary since I only use it once but it seems like a reasonable method nd it looks like it can be used in a few more places in a later refactor. But, I'm fine removing it too.